gum/backend-qnx/gumprocess-qnx.c

```diff
@@ -226,62 +317,72 @@ gum_process_enumerate_malloc_ranges (GumFoundMallocRangeFunc func,
                                      gpointer user_data)
 {
   /* Not implemented */
+  g_assert_not_reached ();
 }
 
 void
 gum_process_enumerate_modules (GumFoundModuleFunc func,
                                gpointer user_data)
 {
-  gchar * as_path;
   gint fd, res;
   gboolean carry_on = TRUE;
   procfs_mapinfo * mapinfos;
   gint num_mapinfos;
-  procfs_debuginfo * debuginfo;
   gint i = 0;
+  GumDlPhdrInternal ** handle;
+  GumDlPhdrInternal * phdr;
 
-  as_path = g_strdup_printf ("/proc/%d/as", getpid ());
-  fd = open (as_path, O_RDONLY);
+  fd = open ("/proc/self/as", O_RDONLY);
   g_assert (fd != -1);
-  g_free (as_path);
 
   res = devctl (fd, DCMD_PROC_PAGEDATA, 0, 0, &num_mapinfos);
-  g_assert (res != 0);
+  g_assert (res == 0);
+
+  if (num_mapinfos == 0)
+    return;
 
   mapinfos = g_malloc (sizeof (procfs_mapinfo) * num_mapinfos);
-  debuginfo = g_malloc (sizeof (procfs_debug) + 0x100);
 
-  res = devctl (fd, DCMD_PROC_PAGEDATA, &mapinfos,
-      sizeof (procfs_mapinfo) * num_mapinfos, NULL);
-  g_assert (res != 0);
+  res = devctl (fd, DCMD_PROC_PAGEDATA, mapinfos,
+      sizeof (procfs_mapinfo) * num_mapinfos, &num_mapinfos);
+  g_assert (res == 0);
 
-  while (carry_on && i != num_mapinfos)
+  handle = dlopen (NULL, RTLD_NOW);
+
+  for (i = 0;
+       carry_on && i != num_mapinfos;
+       i++)
```

Comment:
Sorry if I was unclear, but there's no need to break this into three lines (I only meant for-loops that exceed 80 cols).